### PR TITLE
[AIRFLOW-762] Add Google DataProc delete operator

### PR DESCRIPTION
**Pull request by alexvanboxel**
Pair the recently added Create operator with a Delete operator for
Google Cloud DataProc clusters.

Dear Airflow Maintainers,

Please accept this PR that addresses the following issues:
https://issues.apache.org/jira/browse/AIRFLOW-762

Testing Done:
Integration tests added to the Google Cloud Smoke tests

### CODE REVIEW

**Comment by alexvanboxel** [comment]
@bolkedebruin please can it be included before the beta. 0 risk for functionality and it pairs with a recently added Google Cloud feature. Making it complete.

Integration tests are available at https://github.com/alexvanboxel/airflow-gcp-examples

File: airflow/contrib/operators/dataproc_operator.py [hunk 1]
```diff
@@ -270,6 +270,78 @@ def execute(self, context):
...
+        )
+        service = hook.get_conn()
+
+        response = service.projects().regions().clusters().delete(
+            projectId=self.project_id,
+            region=self.region,
+            clusterName=self.cluster_name
+        ).execute()
+        operation_name = response['name']
+        logging.info("Cluster delete operation name: " + operation_name)
```
Comment (bolkedebruin) [inline airflow/contrib/operators/dataproc_operator.py]: Please use .format

**Comment by bolkedebruin** [comment]
Small nit, if fixed then yes.

**Comment by alexvanboxel** [comment]
PR updated.

**Comment by bolkedebruin** [comment]
+1

**Pull request closed without merging**